src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1521,9 +1521,18 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                     VisitRvalue(methodGroup.ReceiverOpt);
                     LeaveRegionIfNeeded(methodGroup);
                 }
-                else if (node.MethodOpt?.OriginalDefinition is LocalFunctionSymbol localFunc)
+                else if (node.MethodOpt?.OriginalDefinition is { } orgDef)
                 {
-                    VisitLocalFunctionUse(localFunc, node.Syntax, isCall: false);
+                    if (orgDef is LocalFunctionSymbol localFunc)
+                    {
+                        VisitLocalFunctionUse(localFunc, node.Syntax, isCall: false);
+                    }
+                    else if (orgDef is MethodSymbol me && me.IsExtensionMethod)
```

Comment:
Perhaps it would make more sense to instead change the check on line 1518 to include `(node.MethodOpt.RequiresInstanceReceiver || node.MethodOpt.IsExtensionMethod)`.